### ISSUE TITLE
Enhancement: Add I2C interpreter

### ISSUE THREAD

**Issue by nkolban** [nkolban/ESP32_Explorer#2]
Add a tool/web page where the user can submit I2C commands for execution and see the results.

**Comment by shcreasey** [comment]
This sounds interesting.  Could you elaborate a little please?

**Comment by nkolban** [comment]
The ESP32 Explorer tool is a package a number of us are tinkering with.  It provides a "Web Page" interface to a running ESP32 where we can interrogate the ESP32 at runtime.  We can do things like examine storage, examine file systems, toggle GPIOs and much more.  One of the things that might be possible would be to "drive" I2C through either a scripting language or JSON data.  A colleague 
@chegewara , had made a big start on this ... however I haven't had a chance to look at it yet.  Hopefully he will have a comment soon.

**Comment by chegewara** [comment]
This part is just start project and ive moved to implement BT functionality. As for now should work i2c scan, read and write. Read and write i2c bus needs a little more work cause as for now it works only with decimal values and working with i2c is more comfort when we can play with hex

**Comment by shcreasey** [comment]
This is a really cool project.  I think it is going to be very useful.

I haven't managed to get my browser connected to it though :(

**Comment by chegewara** [comment]
Interface to connect to esp32 with flashed ESP32Explorer is in this folder >> https://github.com/nkolban/ESP32_Explorer/tree/master/fatfs_data. 
There is few requirements:
- your local network needs to be in subnet 192.168.1.0,
- address of esp32 need to be 192.169.1.99 

If any of requirement cant be meat you may have to change this line >>>https://github.com/nkolban/ESP32_Explorer/blob/master/fatfs_data/ESP32Explorer.js#L2
and this one: https://github.com/nkolban/ESP32_Explorer/blob/master/main/main.cpp#L43

Also remember to setup wifi ssid ans password. You may have some issue with crash app then open new issue and will handle it.

**Comment by shcreasey** [comment]
Hi Che,
Thanks for your help, I've made a bit of progress.
Does the fatfs_data get loaded onto esp flash automatically when flashing the application?
If so, what's the path to the homepage?  '192.168.1.99/ESP32Explorer.html' doesn't seem to work.

On the other hand, if you are saying I need to open the html page locally in my browser (which I've also tried) how do you then get it to talk to the esp32?

Thanks,
Steve.

**Comment by chegewara** [comment]
No, it is not loaded onto esp32 at all. At the moment im hosting it from my home server, but like i said, you can copy it somwhere onto your computer and open ESP32Explorer.html in browser. Essencially this is index page. 

If you open that html file in browser it will send REST requests straight to the esp32 because in javascript file all ajax requests are hardcoded to 192.168.1.99 aaddress. Thats why those requirements. This probably will change in the future. If you have problem with opening files in browser from local directory just try this http://ns1.diskstation.eu/esp32/

**Comment by shcreasey** [comment]
Great!
Thanks Che :)
It's working (mostly)  now!
(I've raised a new issue relating to file upload)
Regards,
Steve.

**Comment by chegewara** [comment]
As you know its still work in progress. Ive seen issue and working on it now.

**Comment by shcreasey** [comment]
Fab!  (I wasn't complaining, just asking the question / letting you know)
Great project, you guys are doing an amazing job :)

**Comment by chegewara** [comment]
Im glad you like it. If there something you would like to add to functionality or something is not clear just open issue.
